### PR TITLE
Update issue search methods

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/IssueSearchService.java [hunk 1]
```diff
@@ -41,18 +41,25 @@ public IssueSearchService(final JiraCloudService jiraCloudService) {
         this.jiraCloudService = jiraCloudService;
     }
 
-    public IssueSearchResponseModel findIssuesByComment(final String comment) throws IntegrationException {
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/IssueSearchService.java]: We could leave this method as well just to allow further uses if we need it.

File: src/main/java/com/synopsys/integration/jira/common/cloud/model/request/IssueSearchRequestModel.java [hunk 2]
```diff
@@ -89,11 +87,11 @@ public Boolean getFieldsByKeys() {
         return fieldsByKeys;
     }
 
-    private String initialValidationStrategy(final QueryValidationStrategy validationStrategy) {
+    private String initialValidationStrategy(QueryValidationStrategy validationStrategy) {
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/jira/common/cloud/model/request/IssueSearchRequestModel.java]: Why are we removing final? was this a weird formatting change?

File: src/main/java/com/synopsys/integration/jira/common/cloud/model/request/IssueSearchRequestModel.java [hunk 3]
```diff
@@ -89,11 +87,11 @@ public Boolean getFieldsByKeys() {
...
         return null != validationStrategy ? validationStrategy.toString() : QueryValidationStrategy.STRICT.toString();
     }
 
-    private List<String> initialTypesToExpand(final List<ExpandableTypes> typesToExpand) {
+    private List<String> initialTypesToExpand(List<ExpandableTypes> typesToExpand) {
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/jira/common/cloud/model/request/IssueSearchRequestModel.java]: Why are we removing final? was this a weird formatting change?

File: src/main/java/com/synopsys/integration/jira/common/cloud/model/request/IssueSearchRequestModel.java (hunk 2 above)
Comment (gkillough) [reply src/main/java/com/synopsys/integration/jira/common/cloud/model/request/IssueSearchRequestModel.java]: @ekerwin says there is no longer a benefit to using it in method parameters and we should remove it wherever possible.

File: src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/IssueSearchService.java (hunk 1 above)
Comment (gkillough) [reply src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/IssueSearchService.java]: We have the raw JQL method if we want to do things outside of these constraints.

**Pull request merged**